app/models/concerns/publishable.rb

```diff
@@ -20,6 +20,10 @@ module Publishable
       where condition
     }
 
+    before_validation -> {
+      self.published_on ||= Time.zone.now if published?
```

Comment:
It is better to preserve timezone information when writing to DB, but I just checked the tables and it appears that "timestamp without time zone" is used for datetime columns, so it should not matter if we use `Time.now` or `Time.zone.now`.